cognite/client/_cognite_client.py

```diff
@@ -111,3 +112,45 @@ def config(self) -> ClientConfig:
             ClientConfig: The configuration object.
         """
         return self._config
+
+    @classmethod
+    def default(
+        cls,
+        project: str,
+        cdf_cluster: str,
+        tenant_id: str,
+        client_id: str,
+        client_secret: str | None = None,
```

Comment:
Let's make this accept a credential provider instead, and rather rely on factory methods for those to ease instantiation. We can rather add `default_oauth_interactive` and `default_oauth_client_credentials` in addition.